keras_nlp/models/xlm_roberta/xlm_roberta_masked_lm_preprocessor.py

```diff
@@ -0,0 +1,140 @@
+class XLMRoBERTaMaskedLMPreprocessor(XLMRobertaPreprocessor):
+    """XLMRoBERTa preprocessing for the masked language modeling task.
+    This preprocessing layer will prepare inputs for a masked language modeling
+    task. It is primarily intended for use with the
+    `keras_nlp.models.XLMRoBERTaMaskedLM` task model. Preprocessing will occur in
+    multiple steps.
+    - Tokenize any number of input segments using the `tokenizer`.
+    - Pack the inputs together using a `keras_nlp.layers.MultiSegmentPacker`.
+       with the appropriate `"[CLS]"`, `"[SEP]"`, `"[SEP]"`, `"[SEP]"` and `"[PAD]"` tokens.
+    - Randomly select non-special tokens to mask, controlled by
+      `mask_selection_rate`.
+    - Construct a `(x, y, sample_weight)` tuple suitable for training with a
+      `keras_nlp.models.XLMRoBERTaMaskedLM` task model.
+    Examples:
+    ```python
+    # Load the preprocessor from a preset.
+    preprocessor = keras_nlp.models.XLMRoBERTaMaskedLMPreprocessor.from_preset(
+        "xlm_roberta_base_en"
+    )
+    # Tokenize and mask a single sentence.
+    sentence = tf.constant("The quick brown fox jumped.")
+    preprocessor(sentence)
+    # Tokenize and mask a batch of sentences.
+    sentences = tf.constant(
+        ["The quick brown fox jumped.", "Call me Ishmael."]
+    )
+    preprocessor(sentences)
+    # Tokenize and mask a dataset of sentences.
+    features = tf.constant(
+        ["The quick brown fox jumped.", "Call me Ishmael."]
+    )
+    ds = tf.data.Dataset.from_tensor_slices((features))
+    ds = ds.map(preprocessor, num_parallel_calls=tf.data.AUTOTUNE)
+    # Alternatively, you can create a preprocessor from your own vocabulary.
+    # The usage is exactly the same as above.
+    vocab = ["[PAD]", "[UNK]", "[CLS]", "[SEP]", "[MASK]"]
+    vocab += ["THE", "QUICK", "BROWN", "FOX"]
+    vocab += ["Call", "me", "Ishmael"]
+    tokenizer = keras_nlp.models.XLMRoBERTaTokenizer(vocabulary=vocab)
+    preprocessor = keras_nlp.models.XLMRoBERTaMaskedLMPreprocessor(tokenizer)
+    ```
+    """
+
+    def __init__(
+        self,
+        tokenizer,
+        sequence_length=512,
+        truncate="round_robin",
+        mask_selection_rate=0.15,
+        mask_selection_length=96,
+        mask_token_rate=0.8,
+        random_token_rate=0.1,
+        **kwargs,
+    ):
+        super().__init__(
+            tokenizer,
+            sequence_length=sequence_length,
+            truncate=truncate,
+            **kwargs,
+        )
+
+        self.masker = MaskedLMMaskGenerator(
+            mask_selection_rate=mask_selection_rate,
+            mask_selection_length=mask_selection_length,
+            mask_token_rate=mask_token_rate,
+            random_token_rate=random_token_rate,
+            vocabulary_size=tokenizer.vocabulary_size(),
+            mask_token_id=tokenizer.mask_token_id,
+            unselectable_token_ids=[
+                tokenizer.cls_token_id,
```

Comment:
`XLMRobertaTokenizer` class does not have these attributes.
Please replace `cls_token_id` with `start_token_id` and `sep_token_id` with `end_token_id`